### DISCUSSION TITLE
shap with CNN

### DISCUSSION THREAD

**Discussion by alecarlo** [shap/shap#3140] [Q&A]
Hi folk, 
I am pretty new to SHAP.
Is there a way to use SHAP to interpret CNN for regression when both target and predictors are matrices? (i.e., data have two spatial dimensions, lat and lon)

Many thanks!

**Comment by znacer** [comment]
Hi !
You can use `shap.DeepExplainer`. In a similar way of the following example : [PyTorch%20Deep%20Explainer%20MNIST%20example](https://shap.readthedocs.io/en/latest/example_notebooks/image_examples/image_classification/PyTorch%20Deep%20Explainer%20MNIST%20example.html).

**Reply by alecarlo** [reply]
Thanks, but unfortunately shap.DeepExplainer does not support 2D array

AssertionError: The model output must be a vector or a single value!
